lib/boavizta.ts

```diff
@@ -0,0 +1,135 @@
+import {IImpactModelInterface, IStaticParams} from "./interfaces/index";
+import axios from "axios";
+
+export {
+    IImpactModelInterface,
+    IStaticParams
+} from "./interfaces/index";
+
+
+export interface IBoaviztaStaticParams extends IStaticParams {
+    provider?: string;
+    componentType?: string;
+}
+
+export interface IBoaviztaCloudTelemetry {
+
+}
+
+
+export interface IBoaviztaAuthCredentials {
+}
+
+export class BoaviztaCloudImpactModel implements IImpactModelInterface {
+    public provider: string | undefined;
+    public name: string | undefined;
+    protected authCredentials: IBoaviztaAuthCredentials | undefined;
+
+    create(name: string, staticParams: object | undefined): object {
+        const staticParamCast = staticParams as IBoaviztaStaticParams;
+        if (staticParamCast?.hasOwnProperty('provider')) {
+            this.provider = staticParamCast.provider;
+        }
+        this.name = name;
+        return this;
+    }
+
+    authenticate(authParams: object) {
+        this.authCredentials = authParams as IBoaviztaAuthCredentials;
+    }
+
+    createTyped(name: string, staticParams: object | undefined): IImpactModelInterface {
+        const staticParamCast = staticParams as IBoaviztaStaticParams;
+        if (staticParamCast?.hasOwnProperty('provider')) {
+            this.provider = staticParamCast.provider;
+        }
+        this.name = name;
+        return this;
+    }
+
+    async snapshot(data: object): Promise<object> {
+        const dataCast = data as { [key: string]: any };
+        if (!data.hasOwnProperty('provider')) {
+            if (this.provider !== undefined) {
+                dataCast.provider = this.provider;
+            } else {
+                throw new Error('Malformed Telemetry: Missing provider');
+            }
+        }
+        if (!data.hasOwnProperty('instance_type')) {
+            throw new Error('Malformed Telemetry: Missing instance_type');
+        }
+        // let json: any = {"failed": "failed"};
+        // json = new SyncRequestClient()
+        //     .post<object, object>('https://api.boavizta.org/v1/cloud/?verbose=true&allocation=TOTAL', data);
+        const response = await axios.post('https://api.boavizta.org/v1/cloud/', dataCast);
+        return response.data;
+    }
+}
+
+export class BoaviztaComponentImpactModel implements IImpactModelInterface {
```

Comment:
This one looks the same as the other class, is it left here by mistake?

Reply:
Nope. It is intentional.

Reply:
It implements the component impact model which is distinct from cloud impact model.